pallets/ethereum-transactions/src/ethereum_transaction.rs

```diff
@@ -98,31 +101,46 @@ impl DeregisterValidatorData {
 }
 
 #[derive(Encode, Decode, Default, Clone, PartialEq, Debug, Eq, MaxEncodedLen, TypeInfo)]
-pub struct SlashValidatorData {
+pub struct DeregisterCollatorData {
+    pub t1_public_key: H512,
     pub t2_public_key: [u8; 32],
 }
 
-impl SlashValidatorData {
-    pub fn new(t2_public_key: [u8; 32]) -> SlashValidatorData {
-        SlashValidatorData { t2_public_key }
+impl DeregisterCollatorData {
+    pub fn new(t1_public_key: H512, t2_public_key: [u8; 32]) -> DeregisterCollatorData {
+        DeregisterCollatorData { t1_public_key, t2_public_key }
     }
 
     pub fn to_abi(&self) -> EthTransactionDescription {
         EthTransactionDescription {
             function_call: Function {
-                name: String::from("slashValidator"),
-                inputs: vec![Param {
-                    name: String::from("_targetT2PublicKey"),
-                    kind: ParamType::FixedBytes(32),
-                }],
+                name: String::from("deregisterValidator"),
+                inputs: vec![
+                    Param {
+                        name: String::from("_targetT2PublicKey"),
+                        kind: ParamType::FixedBytes(32),
+                    },
+                    Param {
+                        name: String::from("_targetT1PublicKey"),
+                        kind: ParamType::FixedBytes(64),
+                    },
+                ],
```

Comment:
This is the contract we have in T1:
https://github.com/Aventus-Network-Services/avn-bridge/blob/master/contracts/AVNBridge.sol#L381

The order of arguments is:
- T1 public key
- T2 public key
- txId
- confirmations

So this order here is wrong.